Solutions/MarkLogicAudit/Parsers/MarkLogicAudit.txt

```diff
@@ -0,0 +1,61 @@
+// Usage Instructions:
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name as MarkLogicAudit and specifying Legacy Category.
+// This function maps MarkLogic Audit events to Advanced Security Information Model (ASIM) (https://docs.microsoft.com/azure/sentinel/normalization).
+// Function usually takes 10-15 minutes to activate. You can then use function alias from any other queries (e.g. MarkLogicAudit | take 10).
+// Reference : Using functions in Azure monitor log queries : https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+let EventData = MarkLogic_CL
+| extend EventVendor = 'MarkLogic',
+         EventProduct = 'MarkLogic Audit',
+         EventCount = 1;
+let auditlog = EventData
+| where RawData has 'event='
+| mv-apply AdditionalFields = extract_all(@'(?P<key>[a-zA-Z]+)=(?P<value>[a-zA-Z0-9-_:/@.,#{}><" ]+)', dynamic(["key","value"]), RawData) on (
+    project packed1 = pack(tostring(AdditionalFields[0]), tostring(AdditionalFields[1]))
+    | summarize bag = make_bag(packed1)
+)
+| evaluate bag_unpack(bag)
+| extend EventResult = case(success=true, 'Success', 'Failure')
+| project-rename Change=change
+                ,Database=database
+                ,EventType=event
+                ,Expr=expr
+                ,File=file
```

Comment:
Getting the error while executing the parser with the sample data provided. Please add all necessary columns in the sample json provide. Tested with raw file.

![image](https://user-images.githubusercontent.com/13349122/177571515-b4b82ea2-eeba-4b0b-8745-b9a4837abb46.png)

Reply:
@mkchiliveri, added with file column.